module/zfs/zpl_file.c

```diff
@@ -361,8 +361,10 @@ zpl_write(struct file *filp, const char __user *buf, size_t len, loff_t *ppos)
 	ssize_t wrote;
 
 	crhold(cr);
+	spl_inode_lock(file_inode(filp));
 	wrote = zpl_write_common(filp->f_mapping->host, buf, len, ppos,
```

Comment:
Indee, and even in the compat layer the inode ref is taken via the dentry and not the f_mapping. Though it shouldn't really matter. So are you happy with me changing the argument to zpl_write_common to utilise file_inode?